templates/todo/api/java/src/main/resources/application.properties

```diff
@@ -3,11 +3,13 @@ server.port=3100
 spring.jackson.date-format=com.microsoft.azure.simpletodo.configuration.RFC3339DateFormat
 spring.jackson.serialization.WRITE_DATES_AS_TIMESTAMPS=false
 
+# Temporally disabling auto-configuration for key vault
 # Import KeyVault secrets as properties
-spring.cloud.azure.keyvault.secret.property-sources[0].enabled=true
-spring.cloud.azure.keyvault.secret.property-sources[0].endpoint=${AZURE_KEY_VAULT_ENDPOINT}
+# spring.cloud.azure.keyvault.secret.property-sources[0].enabled=true
+# spring.cloud.azure.keyvault.secret.property-sources[0].endpoint=${AZURE_KEY_VAULT_ENDPOINT}
+custom.azure.keyvault.secret.endpoint=${AZURE_KEY_VAULT_ENDPOINT},AZURE-COSMOS-CONNECTION-STRING
```

Comment:
Does this mean that we can no longer launch the app without having `AZURE_KEY_VAULT_ENDPOINT` specified? This used to be that you could only override the KV setting `AZURE_COSMOS_CONNECTION_STRING` locally (pointing to a local emulated DB for example), and the app would still work.

Reply:
I guess we can handle this in our secret resolver to resolve the property locally, if the endpoint if an empty string

Reply:
> Does this mean that we can no longer launch the app without having AZURE_KEY_VAULT_ENDPOINT specified? 

That's currently the case. 
It is true that we can directly set the mongo connection string, but the Key Vault auto config will try to connect anyway to the key vault. 
Currently, the one way to allow manually setting the mongo connection string is by turning `spring.cloud.azure.keyvault.secret.property-sources[0].enabled=false` (to skip key vault connection), then setting `AZURE_COSMOS_CONNECTION_STRING` in the env.

After this PR, the same thing can be done by commenting `custom.azure.keyvault.secret.endpoint=${AZURE_KEY_VAULT_ENDPOINT},AZURE-COSMOS-CONNECTION-STRING` (to disable calling the customize @bean) and enabling `spring.data.mongodb.uri=${AZURE_COSMOS_CONNECTION_STRING:#{null}}`